torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -109,20 +109,28 @@ UserRRef<T>::UserRRef(
 }
 
 template <typename T>
-UserRRef<T>::~UserRRef() {
-  try {
-    RRefContext::getInstance().delUser(ownerId_, rrefId_, forkId_);
-  } catch (const std::exception& ex) {
-    LOG(ERROR) << "Error occurred when deleting UserRRef instance, "
-               << "RRefId = " << rrefId_ << ", ForkId = " << forkId_ << " : "
-               << ex.what();
-  } catch (...) {
-    LOG(ERROR) << "Error occurred when deleting UserRRef instance, "
-               << "RRefId = " << rrefId_ << ", ForkId = " << forkId_ << " : "
-               << "unknown error";
+void UserRRef<T>::tryDel() {
+  bool valid = valid_.exchange(false);
```

Comment:
is it possible to call `tryDel` multiple times, hence this flag? In what cases could that happen?

Reply:
One is on destroy RRefContext, all UserRRefs will have their `tryDel()` called.

The other is when Python references a UserRRef, the destructor of UserRRef will be called. The destructor calls `tryDel`.